lib/dial/middleware.rb

```diff
@@ -63,6 +68,13 @@ def call env
 
     private
 
+    def extract_nonce_from_csp(csp_header)
+      return nil unless csp_header
+
+      nonce_match = csp_header.match(/\'nonce-([^']+)\'/)
+      nonce_match ? nonce_match[1] : nil
```

Comment:
My concern here is using the incorrect nonce with more complex CSPs e.g. what if there are distinct style and script nonces? what if there's only a style nonce and a default of 'self'? what if there are multiple script nonces? etc.